--- a/src/Ouroboros/Network/Socket.hs
+++ b/src/Ouroboros/Network/Socket.hs
@@ -691,7 +691,7 @@ nullNetworkServerTracers :: NetworkServerTracers addr vNumber
 nullNetworkServerTracers = NetworkServerTracers {
-      nstMuxTracer          = nullTracer,
-      nstHandshakeTracer    = nullTracer,
-      nstErrorPolicyTracer  = nullTracer,
-      nstAcceptPolicyTracer = nullTracer
+      nstMuxTracer          = mempty,
+      nstHandshakeTracer    = mempty,
+      nstErrorPolicyTracer  = mempty,
+      nstAcceptPolicyTracer = mempty
     }
