diff --git a/src/Simulation/Network/Snocket.hs b/src/Simulation/Network/Snocket.hs
--- a/src/Simulation/Network/Snocket.hs
+++ b/src/Simulation/Network/Snocket.hs
@@ -56,7 +56,9 @@
 import Control.Monad.Class.MonadTime.SI
 import Control.Monad.Class.MonadTimer.SI
 import Control.Monad.ST.Unsafe (unsafeIOToST)
-import Control.Tracer (Tracer, contramap, contramapM, traceWith)
+import Control.Category ((>>>))
+import Control.Tracer (Tracer (..), contramap, traceWith)
+import qualified Control.Tracer.Arrow as Arrow
 
 import GHC.IO.Exception
 
@@ -86,6 +88,9 @@
 
 import Test.Ouroboros.Network.Data.Script (Script (..), stepScriptSTM)
 
+contramapM :: Monad m => (a -> m b) -> Tracer m b -> Tracer m a
+contramapM f (Tracer tr) = Tracer (Arrow.effect f >>> tr)
+
 data Connection m addr = Connection
     { -- | Attenuated channels of a connection.
       --
